Examples/SmokeTest/SmokeTest/NavigationExamplesView.swift

```diff
@@ -0,0 +1,137 @@
+import SwiftUI
+
+@testable import Honeycomb
+
+// MARK: data models
+struct Park: Identifiable, Equatable, Hashable, Codable {
+    let name: String
+    var id: String {
+        name
+    }
+}
+
+let parks = [
+    Park(name: "Yosemite"),
+    Park(name: "Zion"),
+]
+
+func parkFromId(_ id: Park.ID?) -> Park? {
+    if let parkId = id {
+        if let index = parks.firstIndex(where: { $0.id == parkId }) {
+            return parks[index]
+        }
+    }
+    return nil
+}
+
+struct Tree: Identifiable, Equatable, Hashable, Codable {
+    let name: String
+    var id: String {
+        name
+    }
+}
+
+let trees = [
+    Tree(name: "Oak Tree"),
+    Tree(name: "Maple Tree"),
+]
+
+func treeFromId(_ id: Tree.ID?) -> Tree? {
+    if let treeId = id {
+        if let index = trees.firstIndex(where: { $0.id == treeId }) {
+            return trees[index]
+        }
+    }
+    return nil
+}
+
+// MARK: views
+struct ParkDetails: View {
+    let park: Park
+    var body: some View {
+        Text("details for \(park.name)")
+    }
+}
+
+struct TreeDetails: View {
+    let park: Park
+    let tree: Tree
+    var body: some View {
+        Text("\(park) has many \(tree)")
+    }
+}
+
+struct NavigationStackExample: View {
+    @State private var presentedParks: [Park] = []
+    //  @State private var presentedParks = NavigationPath()
+
+    var body: some View {
+        NavigationStack(path: $presentedParks) {
+            List(parks) { park in
+                NavigationLink(park.name, value: park)
+            }
+            .navigationDestination(for: Park.self) { park in
+                ParkDetails(park: park)
+            }
+        }
+        .instrumentNavigations(path: presentedParks)
+    }
+}
+
+struct NavigationSplitExample: View {
+    @State private var selectedPark: Park.ID? = nil
+    @State private var selectedTree: Park.ID? = nil
+
+    var body: some View {
+        NavigationSplitView {
+            List(parks, selection: $selectedPark) { park in
+                Text(park.name)
+            }
+            .onAppear {
+                reportNavigation(path: "Split View Parks Root")
```

Comment:
As a shape, this seems fine to me? I hear what you're saying about adding a helper that does both the `onAppear` and the `reportNavigation`, but would that cause problems if they add their own `onAppear` before or after our modifier?

I'm also not sure about `reportNavigation` as a top-level name, because I'm worried it's too generic. I mean, like, Sentry has `.sentryTrace()` and our instrumented view name starts with `Honeycomb`. But `reportNavigationToHoneycomb` is pretty ridiculous... I dunno.

Reply:
We could put it on the `Honeycomb` object, so it would be `Honeycomb.reportNavigation`, but that feels wrong to me -- using a class as a namespace.